client/src/hooks.ts

```diff
@@ -11,7 +10,7 @@ export async function handle({
 	resolve,
 }: {
 	event: RequestEvent;
-	resolve(event: RequestEvent): MaybePromise<Response>;
+	resolve(event: RequestEvent): Awaited<Promise<Response>>;
```

Comment:
Typescript loves stacking wrapper types (or whatchamacallit) 😄

Reply:
MaybePromise type was exported from SvelteKit previously, but we moved a to newer version (by package-lock update) and it is now internal only.

Awaited is a fancy new type added in the version of Typescript which we use. Promise was a too strict type for this method.

As far as I understand, this handles a lot of situation better then the strict Promise type. For example when returning raw values from async/await functions (which will be wrapped into a promise by JS), TS will understand that it is a correct. Or when return value is coming from Promise.all(), Promise.race(), etc.
All in all, this is the most safe way to use in TS when we need a type for Promise.

This is the best resource to check features: https://github.com/microsoft/TypeScript/pull/45350